.upstream-tests/test/std/utilities/meta/meta.trans/meta.trans.other/result_of.fail.cpp

```diff
@@ -0,0 +1,42 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+// Mandates: invoke result must fail to compile when used with device lambdas.
+// UNSUPPORTED: !nvcc
+
+// <cuda/std/functional>
+
+// result_of<Fn(ArgTypes...)>
+
+#include <cuda/std/type_traits>
+#include <cuda/std/cassert>
+#include "test_macros.h"
+
+
+template <class Ret, class Fn>
+__host__ __device__
+void test_lambda(Fn &&)
+{
+    ASSERT_SAME_TYPE(Ret, typename cuda::std::result_of<Fn()>::type);
+
+#if TEST_STD_VER > 11
+    ASSERT_SAME_TYPE(Ret, typename cuda::std::invoke_result<Fn>::type);
+#endif
+}
+
+int main(int, char**)
+{
+#if defined(__NVCC__) 
+    { // extended device lambda
```

Comment:
It occurs to me that the files using `__device__` lambdas will fail to compile for the wrong reasons if the `--expt-extended-lambda` flag isn't specified. I don't believe the test compilation will currently add that flag. I don't think we want to add that flag for all test TUs either. I'm not sure if it's possible to add flags with `lit` to only specific test files.

Not sure what the right thing to do is here. I'll defer to @griwes and @wmaxey

Reply:
Judging from the configuration:
```
if self.cxx.type == 'nvcc':                                             
   self.cxx.compile_flags += ['--extended-lambda'] 
```
and the cmake files, the extended lambdas are enabled unconditionally.

Reply:
Oh, huh, I guess we're already doing that. I wonder if that's intentional?

Reply:
Testing with extended device lambdas was intentional.